lock/lock.go

```diff
@@ -0,0 +1,333 @@
+		TableName:            aws.String(options.LockTable),
+		BillingMode:          aws.String(dynamoDbPayPerRequestBillingMode),
+		AttributeDefinitions: attributeDefinitions,
+		KeySchema:            keySchema,
+	})
+
+	if err != nil {
+		if isTableAlreadyBeingCreatedOrUpdatedError(err) {
+			options.Logger.Infof("Looks like someone created table %s at the same time. Will wait for it to be in active state.\n", options.LockTable)
+		} else {
+			return errors.WithStackTrace(err)
+		}
+	}
+
+	return waitForTableToBeActive(options, client)
+}
+
+// Return true if the given error is the error message returned by AWS when the resource already exists and is being
+// updated by someone else
+func isTableAlreadyBeingCreatedOrUpdatedError(err error) bool {
+	awsErr, isAwsErr := err.(awserr.Error)
+	return isAwsErr && awsErr.Code() == "ResourceInUseException"
+}
+
+// Wait for the given DynamoDB table to be in the "active" state. If it's not in "active" state, sleep for the
+// specified amount of time, and try again, up to a maximum of maxRetries retries.
+func waitForTableToBeActive(options *Options, client *dynamodb.DynamoDB) error {
+	return retry.DoWithRetry(options.Logger, fmt.Sprintf("Waiting for Table %s to be active...\n", options.LockTable), maxRetriesWaitingForTableToBeActive, sleepBetweenTableStatusChecks,
+		func() error {
+			isReady, err := lockTableExistsAndIsActive(options.LockTable, client)
+			if err != nil {
+				return err
+			}
+
+			if isReady {
+				options.Logger.Infof("Success! Table %s is now in active state.\n", options.LockTable)
+				return nil
+			}
+
+			return TableNotActiveError{options.LockTable}
+		},
+	)
+}
+
+// Return true if the lock table exists in DynamoDB and is in "active" state
+func lockTableExistsAndIsActive(tableName string, client *dynamodb.DynamoDB) (bool, error) {
+	output, err := client.DescribeTable(&dynamodb.DescribeTableInput{TableName: aws.String(tableName)})
+	if err != nil {
+		if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == "ResourceNotFoundException" {
+			return false, nil
+		} else {
+			return false, errors.WithStackTrace(err)
+		}
+	}
+
+	return *output.Table.TableStatus == dynamodb.TableStatusActive, nil
+}
+
+// GetLockStatus attempts to acquire the lock and check if the expected item is there. If it is - the status is `locked`,
+// if the item with the `LockString` is not there, then the status is `not locked`.
+func GetLockStatus(options *Options) (*dynamodb.GetItemOutput, error) {
+	client, err := NewDynamoDb(options.AwsRegion)
+	if err != nil {
+		options.Logger.Errorf("Error authenticating to AWS: %s\n", err)
+		return nil, err
+	}
+
+	getItemParams := &dynamodb.GetItemInput{
+		Key: map[string]*dynamodb.AttributeValue{
+			attributeLockId: {S: aws.String(options.LockString)},
+		},
+		TableName: aws.String(options.LockTable),
+	}
+
+	item, err := client.GetItem(getItemParams)
+	if err != nil {
+		options.Logger.Errorf("Error authenticating to AWS: %s\n", err)
```

Comment:
Wrong error text

Reply:
oops - fixed now. I'll add some more details to it too